src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -1154,15 +1153,16 @@ internal SingleLookupResult CheckViability(Symbol symbol, int arity, LookupOptio
                                         ref useSiteDiagnostics,
                                         basesBeingResolved))
             {
-                if (inaccessibleViaQualifier)
-                {
-                    diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadProtectedAccess, unwrappedSymbol, accessThroughType, this.ContainingType) : null;
-                }
-                else
-                {
-                    var unwrappedSymbols = ImmutableArray.Create<Symbol>(unwrappedSymbol);
-                    diagInfo = diagnose ? new CSDiagnosticInfo(ErrorCode.ERR_BadAccess, new[] { unwrappedSymbol }, unwrappedSymbols, additionalLocations: ImmutableArray<Location>.Empty) : null;
-                }
+                 bool friendRefNotEqualToThis = getFriendRefNotEqualToThis();
```

Comment:
Calling it up front means we still pay for the execution. I would inline this call.